File: core/src/main/java/com/victorlh/android/framework/core/activitys/AbstractActivity.java
package com.victorlh.android.framework.core.activitys;

import android.content.Context;
import android.content.res.Configuration;
import android.content.res.Resources;
import android.os.Bundle;

import androidx.annotation.AnimRes;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.victorlh.android.framework.core.flujo.ITransitionStackAnimation;
import com.victorlh.android.framework.core.fragment.AbstractFragment;
import com.victorlh.android.framework.core.idioma.IIdioma;

import java.util.Locale;

/**
 * @author Victor
 * 02/06/2018
 */
public abstract class AbstractActivity extends AppCompatActivity {

	protected Context context;

	public AbstractActivity(int contentLayoutId) {
		super(contentLayoutId);
	}

	@Override
	protected void onCreate(@Nullable Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		this.context = this;
		establecerIdioma();
	}

	protected abstract IIdioma getIdiomaAplicacion();

	public void setFragment(int layout, AbstractFragment fragment) {
		setFragment(layout, fragment, null);
	}

	public void setFragment(int layout, AbstractFragment fragment, @Nullable ITransitionStackAnimation animation) {
		if (animation != null) {
			setFragment(layout, fragment, animation.getXmlEntrada(), animation.getXmlSalida());
		} else {
			setFragment(layout, fragment, 0, 0);
		}
	}

	public void setFragment(int layout, AbstractFragment fragment, @AnimRes int xmlEntrada, @AnimRes int xmlSalida) {
		if (fragment == null) {
			fragment = new AbstractFragment() {
			};
		}
		FragmentManager fragmentManager = getSupportFragmentManager();
		FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
		if (xmlEntrada != 0 && xmlSalida != 0) {
			fragmentTransaction.setCustomAnimations(xmlEntrada, xmlSalida);
		}
		fragmentTransaction.replace(layout, fragment);
		fragmentTransaction.commit();
	}

	private void establecerIdioma() {
		IIdioma eIdioma = getIdiomaAplicacion();
		Locale locale = eIdioma.getLocale();
		Locale.setDefault(locale);
		Resources res = this.getResources();
		Configuration config = new Configuration();
		config.locale = locale;
		res.updateConfiguration(config, res.getDisplayMetrics());
	}

	@Override
	public void onSaveInstanceState(@NonNull Bundle outState) {
		super.onSaveInstanceState(outState);
	}

	@Override
	public void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
		super.onRestoreInstanceState(savedInstanceState);
	}
}

File: core/src/main/java/com/victorlh/android/framework/core/activitys/AbstractActivity.kt
package com.victorlh.android.framework.core.activitys

import android.content.res.Configuration
import android.os.Bundle
import androidx.annotation.AnimRes
import androidx.appcompat.app.AppCompatActivity
import com.victorlh.android.framework.core.flujo.ITransitionStackAnimation
import com.victorlh.android.framework.core.fragment.AbstractFragment
import com.victorlh.android.framework.core.idioma.IIdioma
import java.util.*

/**
 * @author Victor
 * 02/06/2018
 */

abstract class AbstractActivity(contentLayoutId: Int) : AppCompatActivity(contentLayoutId) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        establecerIdioma()
    }

    protected abstract val idiomaAplicacion: IIdioma?

    fun setFragment(layout: Int, fragment: AbstractFragment, tag: String) {
        setFragment(layout, fragment, tag, null)
    }

    fun setFragment(layout: Int, fragment: AbstractFragment, tag: String, animation: ITransitionStackAnimation?) {
        setFragment(layout, fragment, tag, animation?.xmlEntrada ?: 0, animation?.xmlSalida ?: 0)
    }

    fun setFragment(layout: Int, fragment: AbstractFragment, tag: String, @AnimRes xmlEntrada: Int, @AnimRes xmlSalida: Int) {
        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        if (xmlEntrada != 0 && xmlSalida != 0) {
            fragmentTransaction.setCustomAnimations(xmlEntrada, xmlSalida)
        }
        fragmentTransaction.replace(layout, fragment, tag)
        fragmentTransaction.commit()
    }

    private fun establecerIdioma() {
        if (this.idiomaAplicacion != null) {
            Locale.setDefault(this.idiomaAplicacion!!.locale)
            val config = Configuration()
            config.setLocale(this.idiomaAplicacion!!.locale)
            createConfigurationContext(config)
            //TODO - Provar
        }
    }

    public override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
    }

    public override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
    }
}